File: 20170425_DictionaryFromTwoLists_7kyu/DictionaryFromTwoLists.js
function createDict(keys, values){
  var obj = {};
  for (var i = 0; i < keys.length; i++) {
    if (values[i] === undefined) {
      obj[keys[i]] = null;
    }
    else {
      obj[keys[i]] = values[i];
    }
  }
  return obj;
}


/* best practices:

function createDict(keys, values) {
  var obj = {}
  
  for (var i = 0; i < keys.length; i++) {
    obj[keys[i]] = i < values.length ? values[i] : null
  }
  
  return obj
}

*/

File: 20170508_DictionaryFromTwoLists_7kyu_Python/DictionaryFromTwoLists.py
def createDict(keys, values):
    return {keys[i]: (values[i] if i in range(0,len(values)) else None) for i in range(0,len(keys))}

# best practice:

# def createDict(keys, values):
#     while len(keys) > len(values):
#         values.append(None)
    
#     dictionary = dict(zip(keys, values)) 
#     return dictionary